src/shogun/lib/type_case.h

```diff
@@ -115,6 +118,10 @@ namespace shogun
 	{                                                                          \
 	};
 
+        SG_ADD_PRIMITIVE_TYPE(CSGObject*, TYPE::T_SGOBJECT)
+        SG_ADD_PRIMITIVE_TYPE(CKernel*, TYPE::T_CKERNEL)
```

Comment:
@gf712 here too is my try to add objects. This works which allowed me to continue the change but obviously I will have to change it so it will be object type (or similar) and not primitive.

Reply:
Yes, just change it to `SG_ADD_OBJECT`